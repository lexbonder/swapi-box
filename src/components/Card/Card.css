.Card {
  border: 1px solid #4A53FE;
  padding: 10px;
  margin: 10px;
  border-radius: 20px;
  font-size: 14px;
  text-align: center;
  width: 261px;
  color: #fff;
  position: relative;
}

.Card button {
  position: absolute;
  background-color: transparent;
  outline: none;
  right: 5px;
  top: 5px;
  color: #4A53FE;
  font-size: 20px;
  border: none;
}

.Card button:hover {
  text-shadow: #feda4a 0 0 10px ;
}

.Card button:active {
  text-shadow: none;
}

.Card ul {
  text-align: left;
  list-style: circle;
  max-height: 87px;
  overflow: auto;
  border: 1px solid #feda4a;
  border-radius: 20px;
}

.favorite {
  box-shadow: 0px 0px 20px #feda4a;
  border: 1px solid #feda4a;
}

.favorite button {
  color: #feda4a;
}