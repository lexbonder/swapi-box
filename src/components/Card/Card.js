import React from 'react';
import PropTypes from 'prop-types';
import './Card.css'

const Card = ({cardInfo, toggleFavorite, currentFav}) => {
  switch(cardInfo.category){
    case 'people' :
      return renderPeopleCards(cardInfo, toggleFavorite, currentFav);
    case 'vehicles' :
      return renderVehicleCards(cardInfo, toggleFavorite, currentFav);
    case 'planets' :
      return renderPlanetCards(cardInfo, toggleFavorite, currentFav);
    default :
      return;
  }
}

const renderPeopleCards = (cardInfo, toggleFavorite, currentFav) => {
  const {name, species, homeworld, population} = cardInfo;
  return (
    <div className={`Card ${currentFav}`}>
      <button
        onClick={toggleFavorite}
        name={name}
      >
        &#9733;
      </button>
      <h1>{name}</h1>
      <h2>Species: {species}</h2>
      <h2>Homeworld: {homeworld}</h2>
      <h2>Population: {population}</h2>
    </div>
  )
}

const renderVehicleCards = (cardInfo, toggleFavorite, currentFav) => {
  const {name, model, vehicleClass, numOfPassengers} = cardInfo;
  return (
    <div className={`Card ${currentFav}`}>
      <button
        onClick={toggleFavorite}
        name={name}
      >
        &#9733;
      </button>
      <h1>{name}</h1>
      <h2>Model: {model}</h2>
      <h2>Class: {vehicleClass}</h2>
      <h2>Passengers: {numOfPassengers}</h2>
    </div>
  )
}

const renderPlanetCards = (cardInfo, toggleFavorite, currentFav) => {
  const {name, terrain, population, climate, residents} = cardInfo;
  const allResidents = residents.map((resident, index) => <li key={index}><h3>{resident}</h3></li>)

  return (
    <div className={`Card ${currentFav}`}>
      <button
        onClick={toggleFavorite}
        name={name}
      >
        &#9733;
      </button>
      <h1>{name}</h1>
      <h2>Terrain: {terrain}</h2>
      <h2>Population: {population}</h2>
      <h2>Climate: {climate}</h2>
      <h2>Residents: {residents.length}</h2>
      <ul>
        {allResidents}
      </ul>
    </div>
  )
}

const { object, func, string } = PropTypes;

Card.propTypes = {
  cardInfo: object,
  toggleFavorite: func,
  currentFav: string
}

export default Card;