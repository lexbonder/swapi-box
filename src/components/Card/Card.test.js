import React from 'react';
import {shallow} from 'enzyme';
import Card from './Card';

describe('Card', () => {
  const mockPerson = {
    category: 'people',
    name: 'Luke Skywalker',
    species: 'human',
    homeworld: 'Tatooine',
    population: '2000000'
  }
  const mockVehicle = {
    category: 'vehicles',
    name: 'RaceCar',
    model: 'Car',
    vehicleClass: 'Really fast',
    numOfPassengers: '4'
  }
  const mockPlanet = {
    category: 'planets',
    name: 'Tatooine',
    terrain: 'Yes',
    population: '2000000',
    climate: 'Most Unpleasant',
    residents: ['Luke', 'Leah', 'The Other Guy']
  }
  const mockToggleFavorite = jest.fn()

  it('should match the snapshot when given a person object', () => {
    const wrapper = shallow(<Card
      cardInfo={mockPerson}
      toggleFavorite={mockToggleFavorite}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when given a vehicle object', () => {
    const wrapper = shallow(<Card
      cardInfo={mockVehicle}
      toggleFavorite={mockToggleFavorite}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot when given a planet object', () => {
    const wrapper = shallow(<Card
      cardInfo={mockPlanet}
      toggleFavorite={mockToggleFavorite}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleFavorite when the favorite button is clicked', () => {
    const wrapper = shallow(<Card
      cardInfo={mockPerson}
      toggleFavorite={mockToggleFavorite}
    />)

    wrapper.find('button').simulate('click')

    expect(mockToggleFavorite).toHaveBeenCalled()
  })
})