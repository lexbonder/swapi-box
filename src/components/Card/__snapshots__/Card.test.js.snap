// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot when given a person object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardInfo={
        Object {
            "category": "people",
            "homeworld": "Tatooine",
            "name": "Luke Skywalker",
            "population": "2000000",
            "species": "human",
          }
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name="Luke Skywalker"
          onClick={[Function]}
>
          ★
</button>,
        <h1>
          Luke Skywalker
</h1>,
        <h2>
          Species: 
          human
</h2>,
        <h2>
          Homeworld: 
          Tatooine
</h2>,
        <h2>
          Population: 
          2000000
</h2>,
      ],
      "className": "Card undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "★",
          "name": "Luke Skywalker",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "★",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            "human",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          "human",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            "Tatooine",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          "Tatooine",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            "2000000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          "2000000",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name="Luke Skywalker"
            onClick={[Function]}
>
            ★
</button>,
          <h1>
            Luke Skywalker
</h1>,
          <h2>
            Species: 
            human
</h2>,
          <h2>
            Homeworld: 
            Tatooine
</h2>,
          <h2>
            Population: 
            2000000
</h2>,
        ],
        "className": "Card undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "★",
            "name": "Luke Skywalker",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "★",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              "human",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            "human",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              "Tatooine",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            "Tatooine",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              "2000000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            "2000000",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot when given a planet object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardInfo={
        Object {
            "category": "planets",
            "climate": "Most Unpleasant",
            "name": "Tatooine",
            "population": "2000000",
            "residents": Array [
              "Luke",
              "Leah",
              "The Other Guy",
            ],
            "terrain": "Yes",
          }
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name="Tatooine"
          onClick={[Function]}
>
          ★
</button>,
        <h1>
          Tatooine
</h1>,
        <h2>
          Terrain: 
          Yes
</h2>,
        <h2>
          Population: 
          2000000
</h2>,
        <h2>
          Climate: 
          Most Unpleasant
</h2>,
        <h2>
          Residents: 
          3
</h2>,
        <ul>
          <li>
                    <h3>
                              Luke
                    </h3>
          </li>
          <li>
                    <h3>
                              Leah
                    </h3>
          </li>
          <li>
                    <h3>
                              The Other Guy
                    </h3>
          </li>
</ul>,
      ],
      "className": "Card undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "★",
          "name": "Tatooine",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "★",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tatooine",
        },
        "ref": null,
        "rendered": "Tatooine",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Terrain: ",
            "Yes",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Terrain: ",
          "Yes",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            "2000000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          "2000000",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Climate: ",
            "Most Unpleasant",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Climate: ",
          "Most Unpleasant",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Residents: ",
            3,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Residents: ",
          3,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <h3>
                            Luke
              </h3>
</li>,
            <li>
              <h3>
                            Leah
              </h3>
</li>,
            <li>
              <h3>
                            The Other Guy
              </h3>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                Luke
</h3>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke",
              },
              "ref": null,
              "rendered": "Luke",
              "type": "h3",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                Leah
</h3>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Leah",
              },
              "ref": null,
              "rendered": "Leah",
              "type": "h3",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                The Other Guy
</h3>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The Other Guy",
              },
              "ref": null,
              "rendered": "The Other Guy",
              "type": "h3",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name="Tatooine"
            onClick={[Function]}
>
            ★
</button>,
          <h1>
            Tatooine
</h1>,
          <h2>
            Terrain: 
            Yes
</h2>,
          <h2>
            Population: 
            2000000
</h2>,
          <h2>
            Climate: 
            Most Unpleasant
</h2>,
          <h2>
            Residents: 
            3
</h2>,
          <ul>
            <li>
                        <h3>
                                    Luke
                        </h3>
            </li>
            <li>
                        <h3>
                                    Leah
                        </h3>
            </li>
            <li>
                        <h3>
                                    The Other Guy
                        </h3>
            </li>
</ul>,
        ],
        "className": "Card undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "★",
            "name": "Tatooine",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "★",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tatooine",
          },
          "ref": null,
          "rendered": "Tatooine",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              "Yes",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            "Yes",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              "2000000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            "2000000",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              "Most Unpleasant",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            "Most Unpleasant",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents: ",
              3,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents: ",
            3,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <h3>
                                Luke
                </h3>
</li>,
              <li>
                <h3>
                                Leah
                </h3>
</li>,
              <li>
                <h3>
                                The Other Guy
                </h3>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  Luke
</h3>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Luke",
                },
                "ref": null,
                "rendered": "Luke",
                "type": "h3",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  Leah
</h3>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Leah",
                },
                "ref": null,
                "rendered": "Leah",
                "type": "h3",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": <h3>
                  The Other Guy
</h3>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "The Other Guy",
                },
                "ref": null,
                "rendered": "The Other Guy",
                "type": "h3",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot when given a vehicle object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardInfo={
        Object {
            "category": "vehicles",
            "model": "Car",
            "name": "RaceCar",
            "numOfPassengers": "4",
            "vehicleClass": "Really fast",
          }
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name="RaceCar"
          onClick={[Function]}
>
          ★
</button>,
        <h1>
          RaceCar
</h1>,
        <h2>
          Model: 
          Car
</h2>,
        <h2>
          Class: 
          Really fast
</h2>,
        <h2>
          Passengers: 
          4
</h2>,
      ],
      "className": "Card undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "★",
          "name": "RaceCar",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "★",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "RaceCar",
        },
        "ref": null,
        "rendered": "RaceCar",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Model: ",
            "Car",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Model: ",
          "Car",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Class: ",
            "Really fast",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Class: ",
          "Really fast",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Passengers: ",
            "4",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Passengers: ",
          "4",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name="RaceCar"
            onClick={[Function]}
>
            ★
</button>,
          <h1>
            RaceCar
</h1>,
          <h2>
            Model: 
            Car
</h2>,
          <h2>
            Class: 
            Really fast
</h2>,
          <h2>
            Passengers: 
            4
</h2>,
        ],
        "className": "Card undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "★",
            "name": "RaceCar",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "★",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "RaceCar",
          },
          "ref": null,
          "rendered": "RaceCar",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model: ",
              "Car",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model: ",
            "Car",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class: ",
              "Really fast",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class: ",
            "Really fast",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers: ",
              "4",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers: ",
            "4",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
